src/core/core.js

```diff
@@ -442,7 +466,8 @@ Crafty.fn = Crafty.prototype = {
             }
         }
         delete this.__c[id];
-
+        // update map from component to (entityId -> entity)
```

Comment:
> It could be changed to reflect this by having `{ componentName: [id1, ..., idn] }`. 

This would probably be poor perf wise -- it would be expensive to delete ids out of the list.  The minimal approach would be something like `{componentName: {id1: true, id5: true}}` etc, but you're right that we might as well just point to entities instead.

(This is slightly related to #1106, since it's another place that a [set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) might be a slightly better match.).